### PR TITLE
Add configuration file and environment variables.

### PR DESCRIPTION
**Pull request by bgravenorst**
Signed-off-by: Byron Gravenorst <byron.gravenorst@consensys.net>

## Pull Request Description

Add configuration file and environment variables.

Additionally updated the Docker instructions to show an example using the env variables.

### CODE REVIEW

File: docs/Reference/CLI/CLI-Subcommands.md [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+* As [environment variables](#environment-variables)
+* In a TOML configuration file.
+
+For example, you can set the options for the `azure-signer` subcommand in environment variables
+or a configuration file, but the subcommand must be specified in the command line
+
+!!! example
+
+    ```bash
+    ethsigner --config-file=./config.yaml azure-signer
```
Comment (macfarla) [inline docs/Reference/CLI/CLI-Subcommands.md:38]: this says yaml but above says TOML

File: docs/Reference/CLI/CLI-Subcommands.md [hunk 2]
```diff
@@ -0,0 +1,499 @@
+---
+description: Ethsigner command line interface subcommands
```
Comment (macfarla) [inline docs/Reference/CLI/CLI-Subcommands.md:2]: s/Ethsigner/EthSigner

File: docs/HowTo/Get-Started/Use-Docker.md [hunk 3]
```diff
@@ -28,6 +28,14 @@ Display the EthSigner command line help using the Docker image:
 !!! note
     `latest` runs the latest cached version. To pull the latest version, use `docker pull pegasyseng/ethsigner:latest`.
 
+You can specify [environment variables] with the docker image instead of the command line options.
+
+=== "example"
+
+    ```bash
+    docker run -e ETHSIGNER_CONFIG_FILE=/home/me/me_node/config.toml pegasyseng/ethsigner:latest file-based-signer
```
Comment (usmansaleem) [inline docs/HowTo/Get-Started/Use-Docker.md:36]: I need to test it, but I suspect this particular example might need a volume mount i.e. /home/me/me_node should be mounted to a folder inside docker, then the config file env variable can be updated to point to mounted location. Similar to how password files are specified at line 55 `-v <~/myKeyFile>:/opt/ethsigner/keyfile`

File: docs/HowTo/Use-Configuration-File.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+!!! note
+
+    Subcommands cannot be specified in the configuration file, you can only specify a subcommand's
+    associated options. The signing key subcommand must be specified on the command line.
+
+Values must conform to TOML specifications for string, numbers, arrays, and booleans. Specific
+differences between the command line and the TOML file format are:
+
+* Comma-separated lists on the command line are string arrays in the TOML file.
+* Enclose file paths, hexadecimal numbers, URLs, and &lt;host:port> values in quotes.
```
Comment (usmansaleem) [inline docs/HowTo/Use-Configuration-File.md:33]: html escape.

File: docs/Reference/CLI/CLI-Subcommands.md (hunk 1 above)
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Subcommands.md:38]: Done

File: docs/Reference/CLI/CLI-Subcommands.md (hunk 2 above)
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Subcommands.md:2]: Done

File: docs/HowTo/Use-Configuration-File.md (hunk 4 above)
Comment (bgravenorst) [reply docs/HowTo/Use-Configuration-File.md:33]: Done.

File: docs/HowTo/Get-Started/Use-Docker.md (hunk 3 above)
Comment (bgravenorst) [reply docs/HowTo/Get-Started/Use-Docker.md:36]: I'll update this to:

`docker run --mount type=bind,source=/home/me/me_node,target=/var/lib/ethsigner -e ETHSIGNER_CONFIG_FILE=/var/lib/ethsigner/config.toml pegasyseng/ethsigner:latest file-based-signer`

File: docs/HowTo/Get-Started/Use-Docker.md [hunk 5]
```diff
@@ -28,6 +28,14 @@ Display the EthSigner command line help using the Docker image:
...
     `latest` runs the latest cached version. To pull the latest version, use `docker pull pegasyseng/ethsigner:latest`.
 
+You can specify [environment variables] with the docker image instead of the command line options.
+
+=== "example"
+
+    ```bash
+    docker run --mount type=bind,source=/home/me/me_node,target=/var/lib/ethsigner -e ETHSIGNER_CONFIG_FILE=/var/lib/ethsigner/config.toml pegasyseng/ethsigner:latest file-based-signer
+    ```
+
```
Comment (usmansaleem) [inline docs/HowTo/Get-Started/Use-Docker.md:38]: that looks good. 

nit-pick ... this particular example uses config file option from environment variable, does it make sense to have another alternate example to use environment variables without config file? i.e. 

`docker run -p <listenPort>:8545 -v <~/myKeyFile>:/opt/ethsigner/keyfile -v <~/myPasswordFile>:/opt/ethsigner/passwordfile pegasyseng/ethsigner:latest --chain-id=2018 --downstream-http-host=<PantheonHost> --downstream-http-port=8590 --http-listen-host=0.0.0.0 file-based-signer -k /opt/ethsigner/keyfile -p /opt/ethsigner/passwordfile`

can also be written as

`docker run -p <listenPort>:8545 -v <~/myKeyFile>:/opt/ethsigner/keyfile -v <~/myPasswordFile>:/opt/ethsigner/passwordfile -e ETHSIGNER_CHAIN_id=2018 -e ETHSIGNER_DOWNSTREAM_HTTP_HOST=<BesuHost> -e ETHSIGNER_DOWNSTREAM_HTTP_PORT=8590 -e ETHSIGNER_HTTP_LISTEN_HOST=0.0.0.0 -e ETHSIGNER_FILE_BASED_SIGNER_KEY_FILE=/opt/ethsigner/keyfile -e  ETHSIGNER_FILE_BASED_SIGNER_PASSWORD_FILE=/opt/ethsigner/passwordfile pegasyseng/ethsigner:latest file-based-signer `

**Review by usmansaleem — APPROVED** [review]
LGTM - minor suggestion around adding another example for docker and environment variables without using config file.

**Pull request merged**